### PR TITLE
LifeCycleAssessment_oM: Removal of Scope Object classes and Creation of Scope Fragment

### LINKED ISSUE #1367: LifeCycleAssessment_oM: Remove scope objects and revise engine functionality
#### Description:

From a small pool of user feedback, it seems that Scopes as formatted are largely unused and slightly confusing to work with throughout the LCA toolkit. That being said, Scopes as a concept remain important albeit not yet implemented across the BHoM as a whole. The preliminary Scope objects have served as a simple test to help formulate LCA workflows, and it has been made evident that these objects can cause confusion in a standard workflow. It is proposed that the multiple Scope objects which hold "typical discipline objects" be removed and replaced with a `ScopeType` fragment that can be added to any object (or collection of objects) depending on the project needs. Project-specific metadata collected on a workflow has been facilitated by the `LifeCycleAssessmentScope` but this too can also be shifted to a fragment OR a core property on the `ProjectLifeCycleAssessment` depending on the overall intent with metadata (up for discussion). 

`ProjectLifeCycleAssessment` can still be used to gather all of the elements being evaluated, but there's no real need to aggregate objects in the current multi-tiered hierarchy. By simply providing a fragment to the objects themselves, the calculation methods can be much cleaner because we will not have to loop through all of the individual scope objects and scope disciplines prior to solving anything. 

Scope objects to be removed: 
- [x] ElectricalScope
- [x] EnclosuresScope
- [x] fireProtectionScope
- [x] foundationsScope
- [x] IScope
- [x] MechanicalScope
- [x] MEPScope
- [x] PlumbingScope
- [x] StructuresScope
- [x] TenantImprovementScope

Scope objects to be updated: 
- [x] LifeCycleAssessmentScope
- [x] ProjectLifeCycleAsessment

Other objects that will be impacted: 
- [x] EnvironmentalMetricResult
- [x] GlobalWarmingResult
- [x] LifeCycleAssessmentElementResult

Other objects that need to be modified will accompany the PR changelog. 

This will require significant versioning updates and therefore would be beneficial to have the support of @FraserGreenroyd when time comes to correctly version new updates. 

FYI @kayleighhoude @enarhi

### PR DESCRIPTION
**Pull request by michaelhoehn**
Addition of Scope Fragment

### Issues addressed by this PR

Closes #1367 

### Test files

[Test file](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM/LifeCycleAssessment_oM/%231370-Remove-Scope-Objects/%231367-removescopes.gh?csf=1&web=1&e=VifcEY)

### CODE REVIEW

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check project-compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by michaelhoehn** [comment]
*sigh* 
@BHoMBot check versioning

File: LifeCycleAssessment_oM/Versioning_51.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  },
+  "Property": {
+    "ToNew": {
+    },
+    "ToOld": {
+    }
+  },
+  "MessageForDeleted": {
+    "BH.oM.LifeCycleAssessment.ScopeObjects.ElectricalScope": "This object has been replaced with a Scope fragment.",
```
Comment (michaelhoehn) [inline LifeCycleAssessment_oM/Versioning_51.json:23]: @FraserGreenroyd Version report didn't like this very much. Would you be able to help me out with the versioning here?

Comment (FraserGreenroyd) [reply LifeCycleAssessment_oM/Versioning_51.json:23]: The namespace for the objects does not contain `ScopeObjects` - the objects were not compliant by the looks of it on that basis. Removing `ScopeObjects` from the object names here and it should then be happy.

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Review by enarhi — APPROVED** [review]
LGTM

**Comment by michaelhoehn** [comment]
@BHoMBot check required

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check required

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Review by enarhi — APPROVED** [review]
LGTM

**Comment by michaelhoehn** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**